sources/shdr/App.coffee

```diff
@@ -186,6 +186,18 @@ class App
             shdr.UI.WARNING)
         console.warn 'ERROR: ', e
 
+  upload: (file_obj) ->
+    try
+      reader = new FileReader()
+      reader.readAsDataURL file_obj
+      reader.onload = (e) =>
+        model = {name: file_obj.name, data: e.target.result}
+        shdr.Models[e.target.result] = model
+        @ui.setStatus('Uploaded', shdr.UI.SUCCESS)
```

Comment:
Maybe add another status at the beginning to notify the user something is happening, even just `Uploading...`. Since this is all local it's probably fast enough to not really matter, but it's a good pattern to let the user know what is happening in case it is not fast.